src/Microsoft.Restier.Core/ApiBase.cs

```diff
@@ -67,7 +67,14 @@ protected ApiConfiguration ApiConfiguration
                         var builder = this.ConfigureApi(new ApiBuilder());
                         EnableConventions(builder, apiType);
                         ApiConfiguratorAttribute.ApplyApiBuilder(apiType, builder);
-                        builder.TryUseSharedApiScope(); // TODO: Maybe default to context scope?
+                        if (!builder.HasService<IApi>())
+                        {
+                            builder.Services
+                                .AddScoped(apiType, sp => sp.GetService<ApiScope>().Api)
+                                .AddScoped<IApi>(sp => sp.GetService<ApiScope>().Api)
+                                .AddScoped<ApiContext>(sp => sp.GetService<ApiScope>().Api.ApiContext)
```

Comment:
I think we can eliminate the generic parameter since the return type of the lambda is `ApiContext`.